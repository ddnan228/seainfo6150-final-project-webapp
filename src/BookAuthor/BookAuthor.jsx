import React from 'react'
import css from "./BookAuthor.module.css";

const BookAuthor = props => {
  return (
    <div className = {css.authorStyle}>
      <h2 className={css.title}>Author</h2>
      <h2 className={css.authorName}>{props.author}</h2>
      <p className={css.aboutAuthor}>{props.aboutAuthor}</p>
    </div>
  );
};

export default BookAuthor;
