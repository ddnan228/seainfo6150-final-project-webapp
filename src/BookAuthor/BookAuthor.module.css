.authorStyle {
  width: 280px;
  height: 280px;
  margin: 10px;
  border-radius: 15px;
  background-color: #ffcc99;
  text-align: center;
  padding: 20px;
  font-size: 15px;
  color: #4d1f00;
  box-shadow: 5px 5px 10px #ffd9b3;
  border: 5px solid #ffbf80;
}

.title {
  font-family: 'Balsamiq Sans', cursive;
}
.authorName {
  font-family: 'Sansita Swashed', cursive;
}

.aboutAuthor {
  font-family: 'Balsamiq Sans', cursive;
}
