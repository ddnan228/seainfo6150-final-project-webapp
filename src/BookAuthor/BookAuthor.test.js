import React from "react";
import { render } from "@testing-library/react";
import BookAuthor from "./BookAuthor.jsx";

it("renders the BookAuthor component correctly", () => {
  const author="Leslie Patricelli";
  const aboutAuthor="Leslie Patricelli is the author-illustrator of a series of best-selling board books about everybody’s favorite bald baby. She is also the creator of the picture books Faster! Faster!, Bigger! Bigger!, and the Boston Globe–Horn Book Honor Book Higher! Higher!, among other books for children. She lives in Idaho.";

  const { container } = render(<BookAuthor author={author} aboutAuthor={aboutAuthor} />);
  expect(container).toMatchSnapshot();
});
