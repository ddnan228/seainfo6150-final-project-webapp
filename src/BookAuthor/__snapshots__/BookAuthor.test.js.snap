// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the BookAuthor component correctly 1`] = `
<div>
  <div
    class="authorStyle"
  >
    <h2
      class="title"
    >
      Author
    </h2>
    <h2
      class="authorName"
    >
      Leslie Patricelli
    </h2>
    <p
      class="aboutAuthor"
    >
      Leslie Patricelli is the author-illustrator of a series of best-selling board books about everybody’s favorite bald baby. She is also the creator of the picture books Faster! Faster!, Bigger! Bigger!, and the Boston Globe–Horn Book Honor Book Higher! Higher!, among other books for children. She lives in Idaho.
    </p>
  </div>
</div>
`;
